sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -694,12 +694,25 @@ private[hive] class HiveClientImpl(
       tableName: String,
       replace: Boolean,
       isSrcLocal: Boolean): Unit = withHiveState {
-    shim.loadTable(
-      client,
-      new Path(loadPath),
-      tableName,
-      replace,
-      isSrcLocal)
+    val tbl = client.getTable(tableName)
+    val fs = tbl.getDataLocation.getFileSystem(conf)
+    if (replace) {
```

Comment:
[`loadTable` is calling `replaceFiles` when `replace` is true. ](https://github.com/apache/hive/blob/release-1.2.1/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L1640), instead of calling `Hive.copyFiles`.

`replaceFiles` is based on the calls of `moveFile`.